packages/java/endpoint/src/main/java/dev/hilla/crud/PropertyStringFilterSpecification.java

```diff
@@ -148,6 +148,24 @@ public Predicate toPredicate(Root<T> root, CriteriaQuery<?> query,
             default:
                 break;
             }
+        } else if (javaType.isEnum()) {
+            Enum<?> enumValue = parseEnumValue(javaType, value);
```

Comment:
```suggestion
            var enumValue = Enum.valueOf(javaType.asSubclass(Enum.class), value);
```
And then you can remove `parseEnumValue`.